test/runtime_tests.cc

```diff
@@ -213,31 +213,37 @@ namespace detail {
 		}
 	}
 
-	TEST_CASE("task_manager invokes callback upon task creation", "[task_manager]") {
-		task_manager tm{1, nullptr};
-		size_t call_counter = 0;
-		tm.register_task_callback([&call_counter](const task*) { call_counter++; });
+	TEST_CASE("task_manager calls into delegate on task creation", "[task_manager]") {
+		struct counter_delegate : public task_manager::delegate {
+			size_t counter = 0;
+			void notify_task_created(const task* /* tsk */) override { counter++; }
+		};
+
+		counter_delegate delegate;
+		task_manager tm{1, nullptr, &delegate};
+		tm.generate_epoch_task(epoch_action::init);
+		CHECK(delegate.counter == 1);
```

Comment:
This change illustrates pretty well why I'm unconvinced by the whole "delegate" design when it's just about exactly one callback.

Reply:
Well, it's inside a test. I much prefer the delegate when the owner of `task_manager` is another class.